mycore-neo4j/src/main/java/org/mycore/mcr/neo4j/utils/MCRNeo4JQueryRunner.java

```diff
@@ -0,0 +1,235 @@
+                     pathSB.append(start).append(',');
+
+                     pathSB.append(end).append("],");
+                     pathSB.append("\"relationships\":").append(relationshipJson);
+
+                     pathSB.append("}}");
+
+                     keyMap.put("path_" + key, pathSB.toString());
+                  } else if (StringUtils.equals(recordData.type().name(), "NULL")) {
+                     LOGGER.warn("Got record of type {} for key {} which is not parsed and is ignored",
+                           recordData.type().name(), key);
+                  } else {
+                     LOGGER.warn("Got record of type {} for key {} which is not parsed",
+                           recordData.type().name(), key);
+                     keyMap.put(key, gson.toJson(thisRecord.asMap()));
+                  }
+               }
+               records.add(keyMap);
+
+            }
+            return records;
+         });
+      }
+
+   }
+
+   private static String nodeToJson(Node startNode, Gson gson, String lang) {
+      String elementId = startNode.elementId();
+      boolean mcridBool = startNode.asMap().containsKey("id");
+
+      List<Neo4JMetaData> neo4JMetaDataList = propertiesMapToJson(startNode.asMap(), lang);
+
+      JsonObject jsonNode = new JsonObject();
+      JsonArray labelArray = new JsonArray();
+      for (String label: startNode.labels()){
+         labelArray.add(label);
+      }
+      jsonNode.add("type", labelArray);
+      jsonNode.addProperty( "id", elementId);
+      if (mcridBool) {
+         String mcrID = String.valueOf(startNode.asMap().get("id"));
+         LOGGER.debug("MCRID: {}", mcrID);
+         jsonNode.addProperty("mcrid", mcrID);
+      }
+      JsonArray jsonArray = new JsonArray();
+      for (Neo4JMetaData metaData: neo4JMetaDataList) {
+         JsonObject jsO = new JsonObject();
+         jsO.addProperty("title", metaData.title());
+         JsonArray innerArray = new JsonArray();
+         for (String contentString: metaData.content()) {
+            innerArray.add(contentString);
+         }
+         jsO.add("content", innerArray);
+         jsonArray.add(jsO);
+      }
+
+      jsonNode.add("metadata", jsonArray);
+
+      ObjectMapper objectMapper = new ObjectMapper();
+      objectMapper.configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY, true);
+      objectMapper.configure(SerializationFeature.WRITE_SINGLE_ELEM_ARRAYS_UNWRAPPED, true);
+      objectMapper.configure(SerializationFeature.WRAP_ROOT_VALUE, false);
+
+      try {
+         Neo4JNodeJsonRecord node = objectMapper.readValue(gson.toJson(jsonNode), Neo4JNodeJsonRecord.class);
+         return objectMapper.writeValueAsString(node);
+      } catch (JsonProcessingException e) {
+         LOGGER.error(e.getLocalizedMessage());
+      }
+
+      return null;
+   }
+
+   private static List<Neo4JMetaData> propertiesMapToJson(Map<String, Object> map, String lang) {
```

Comment:
Why does name end with 'toJSON?'

Reply:
renamed the function and add a little comentary